django/thunderstore/repository/models/cache.py

```diff
@@ -111,3 +117,166 @@ def drop_stale_cache(cls):
                 entry.delete()
         for entry in cls.objects.filter(community=None).iterator():
             entry.delete()
+
+
+# TODO: unit tests
+class APIV1ChunkedPackageCache(SafeDeleteMixin):
+    community: Community = models.ForeignKey(
+        "community.Community",
+        related_name="chunked_package_list_cache",
+        on_delete=models.CASCADE,
+    )
+    index: DataBlob = models.OneToOneField(
+        "storage.DataBlob",
+        related_name="chunked_package_indexes",
+        on_delete=models.PROTECT,
+    )
+    chunks: DataBlobGroup = models.ForeignKey(
+        "storage.DataBlobGroup",
+        related_name="chunked_package_list_cache",
+        on_delete=models.PROTECT,
+    )
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    class Meta:
+        get_latest_by = "created_at"
+
+    @classmethod
+    def get_latest_for_community(
+        cls,
+        community: Community,
+    ) -> Optional["APIV1ChunkedPackageCache"]:
+        return cls.objects.filter(community=community.pk).latest()
+
+    @classmethod
+    def update_for_community(cls, community: Community) -> None:
+        """
+        Chunk community's PackageListings into blob files and create an
+        index blob that points to URLs of the chunks.
+        """
+        uncompressed_blob_size = 14000000  # 14MB compresses into ~1MB files.
+        group = DataBlobGroup.objects.create(
+            name=f"Chunked package list: {community.identifier}",
+        )
+        chunk_content = bytearray()
+
+        def finalize_blob() -> None:
+            group.add_entry(
+                gzip.compress(b"[" + chunk_content + b"]"),
+                name=f"Package list chunk for {community.identifier}",
+                content_type="application/json",
+                content_encoding="gzip",
+            )
+
+        for listing in get_package_listings(community):
+            listing_bytes = listing_to_json(listing)
+
+            # Start new blob if adding current chuck would exceed the size limit.
+            # +2 for closing brackets
+            if len(chunk_content) + len(listing_bytes) + 2 > uncompressed_blob_size:
+                finalize_blob()
+                chunk_content = bytearray(listing_bytes)
+            else:
+                chunk_content.extend(
+                    listing_bytes if not len(chunk_content) else b"," + listing_bytes,
+                )
+
+        if len(chunk_content):
```

Comment:
TODO: need to finalize an empty chunk if community has no active package listings at all.